frontend/src/lib/logic/featureFlagLogic.ts

```diff
@@ -20,7 +20,7 @@ function notifyFlagIfNeeded(flag: string, flagState: string | boolean | undefine
 
 function getPersistedFeatureFlags(appContext: AppContext | undefined = getAppContext()): FeatureFlagsSet {
     const persistedFeatureFlags = appContext?.persisted_feature_flags || []
-    return Object.fromEntries(persistedFeatureFlags.map((f) => [f, true]))
+    return Object.fromEntries(persistedFeatureFlags.map((f) => (Array.isArray(f) ? f : [f, true])))
```

Comment:
you can now default the `persistedFeatureFlags` to be an object (Record<string, string | boolean>) and get rid of the whole `Object.fromEntries()` schenanigans here and simplify things everywhere else too.

Just add `true` as value to previous default persisted flags